Source/WebCore/page/RemoteDOMWindow.cpp

```diff
@@ -113,13 +114,38 @@ WindowProxy* RemoteDOMWindow::parent() const
     return &m_frame->windowProxy();
 }
 
-void RemoteDOMWindow::postMessage(JSC::JSGlobalObject&, LocalDOMWindow& incumbentWindow, JSC::JSValue message, const String& targetOrigin, Vector<JSC::Strong<JSC::JSObject>>&&)
-{
-    UNUSED_PARAM(incumbentWindow);
-    UNUSED_PARAM(message);
-    UNUSED_PARAM(targetOrigin);
 
-    // FIXME: Implemented this.
+void RemoteDOMWindow::postMessage(JSC::JSGlobalObject& lexicalGlobalObject, LocalDOMWindow& incumbentWindow, JSC::JSValue message, const String& targetOrigin, Vector<JSC::Strong<JSC::JSObject>>&& transfer)
+{
+    RefPtr sourceDocument = incumbentWindow.document();
+    // Compute the target origin. We need to do this synchronously in order
+    // to generate the SyntaxError exception correctly.
+    std::optional<SecurityOriginData> target;
+    if (targetOrigin == "/"_s) {
+        if (!sourceDocument)
+            return;
+        target = sourceDocument->securityOrigin().data();
+    } else if (targetOrigin != "*"_s) {
+        target = SecurityOrigin::createFromString(targetOrigin)->data();
+        // It doesn't make sense target a postMessage at an opaque origin
+        // because there's no way to represent an opaque origin in a string.
+        if (target->isOpaque())
+            return; // Exception { SyntaxError };
```

Comment:
This should still return a SyntaxError, and this whole function should return ExceptionOr<void>, and we should have a test that tries to post a message to a RemoteDOMWindow with an explicit opaque origin and verify that it throws the exception in JS with site isolation enabled like it does with site isolation not enabled.